src/decorated_surface.rs

```diff
@@ -355,11 +384,23 @@ impl<H: Handler> DecoratedSurface<H> {
     /// Sets this surface as fullscreen (see `wl_shell_surface` for details)
     ///
     /// Automatically disables decorations.
-    pub fn set_fullscreen(&mut self,
+    ///
+    /// Note: When using `xdg_shell`, the requested `FullscreenMethod` and `framerate` will be
+    /// ignored. Not sure if some way of handling these arguments should be added to the xdg_shell?
```

Comment:
Given the functionality is not supported by xdg_shell and wl_shell is supposed to be deprecated anyway, we may simply remove it, and use the default values `FullscreenMethod::Default` for `method` and `0` for `framerate` (which together means "let the server decide", so exactly what xdg_shell does).